packages/browser-wallet/src/popup/pages/AddTokens/ExternalAddTokens.tsx

```diff
@@ -0,0 +1,237 @@
+    };
+
+    const contractDetails = useAsyncMemo(
+        async () => {
+            const name = await fetchContractName(client, BigInt(contractIndex), BigInt(contractSubindex));
+            if (!name) {
+                throw new Error('Test');
+            }
+            return { index: BigInt(contractIndex), subindex: BigInt(contractSubindex), contractName: name };
+        },
+        (e) => addToast(e.message),
+        []
+    );
+
+    useEffect(() => {
+        if (!contractDetails || accountTokens.loading) {
+            return;
+        }
+        const existingIds = (accountTokens.value[contractIndex] || []).map((token) => token.id);
+        fetchTokenMetadata(tokenIds, existingIds, client, network, contractDetails).then((newTokens) =>
+            setAddingTokens(newTokens)
+        );
+    }, [contractDetails, accountTokens.loading]);
+
+    if (!contractDetails || !addingTokens) {
+        return null;
+    }
+
+    const updateTokenStatus = (index: number) => {
+        const token = addingTokens[index];
+        const status = token.status === ChoiceStatus.chosen ? ChoiceStatus.discarded : ChoiceStatus.chosen;
+        addingTokens[index] = { ...token, status };
+        setAddingTokens([...addingTokens]);
+    };
+
+    const allExisting = addingTokens.every(({ status }) => status === ChoiceStatus.existing);
+
+    return (
+        <ExternalRequestLayout>
+            <ConnectedBox accountAddress={account} url={new URL(url).origin} />
+            <div className="h-full flex-column align-center">
+                <p>
+                    {t(allExisting ? 'prompt.descriptionAllExisting' : 'prompt.description', { dApp: displayUrl(url) })}
+                </p>
+                <UncontrolledInput
+                    readOnly
+                    className="add-tokens-prompt__input w-full"
+                    label={t('contractName')}
+                    value={contractDetails.contractName}
+                />
+                <div className="add-tokens-prompt__token-container w-full">
+                    {addingTokens.map((token, index) => (
+                        <DisplayToken
+                            key={token.id}
+                            status={token.status}
+                            metadata={token.metadata}
+                            balance={0n}
+                            updateStatus={() => updateTokenStatus(index)}
+                        />
+                    ))}
+                </div>
+                <div className="flex p-b-10  m-t-auto">
+                    {allExisting || (
+                        <>
+                            <Button
+                                width="narrow"
+                                className="m-r-10"
+                                onClick={withClose(() => respond(addingTokens.filter(onlyExisting).map(getId)))}
+                            >
+                                {t('prompt.reject')}
+                            </Button>
```

Comment:
I don't think we should reveal this information when the user rejects. Seems misleading. How do we signal rejection in other prompts?

Reply:
I'll just do the empty list instead, like I do with the onClose.